app/src/pages/SingleContent.vue

```diff
@@ -22,9 +23,28 @@ const content = db.getBySlugAsRef<ContentDto>(props.slug);
 const tagsContent = ref<ContentDto[]>([]);
 
 watch(
-    content,
+    [content, appLanguageAsRef],
     async () => {
-        if (!content.value) return;
+        if (!content.value) {
+            return;
+        }
+
+        if (appLanguageAsRef.value?._id != content.value.language) {
+            const contentDocs = await db.whereParent(content.value.parentId);
+            const preferred = contentDocs.find((c) => c.language == appLanguageAsRef.value?._id);
+
+            if (preferred) {
+                await router.push({ name: "content", params: { slug: preferred.slug } });
+                content.value = preferred;
```

Comment:
This is needed because routing to the same route will not remount the component but keep using the same instance. For clarity, I would suggest to place this one line up, above the `router` call.